src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/AddUsingsCodeActionResolver.cs

```diff
@@ -0,0 +1,239 @@
+            {
+                return null;
+            }
+
+            var codeDocument = await document.GetGeneratedOutputAsync().ConfigureAwait(false);
+            if (codeDocument.IsUnsupported())
+            {
+                return null;
+            }
+
+            if (!FileKinds.IsComponent(codeDocument.GetFileKind()))
+            {
+                return null;
+            }
+
+            var codeDocumentIdentifier = new VersionedTextDocumentIdentifier() { Uri = actionParams.Uri, Version = 0 };
+            var documentChanges = new List<WorkspaceEditDocumentChange>();
+
+            var namespaceList = actionParams.Namespaces;
+
+            var usingDirectives = FindUsingDirectives(codeDocument);
+            if (usingDirectives.Count > 0)
+            {
+                documentChanges.Add(GenerateUsingEditsInterpolated(codeDocument, codeDocumentIdentifier, actionParams.Namespaces, usingDirectives));
+            }
+            else
+            {
+                documentChanges.Add(GenerateUsingEditsAtTop(codeDocument, codeDocumentIdentifier, actionParams.Namespaces));
+            }
+
+            return new WorkspaceEdit()
+            {
+                DocumentChanges = documentChanges
+            };
+        }
+
+        private static WorkspaceEditDocumentChange GenerateUsingEditsInterpolated(
+            RazorCodeDocument codeDocument,
+            VersionedTextDocumentIdentifier codeDocumentIdentifier,
+            string[] namespaceArray,
+            List<RazorUsingDirective> usingDirectives)
+        {
+            // Sort and queue namespaces
+            var namespaceList = namespaceArray.ToList();
+            namespaceList.Sort();
+            var namespaces = new Queue<string>(namespaceList);
+
+            var edits = new List<TextEdit>();
+            foreach (var usingDirective in usingDirectives)
+            {
+                // Break early if we're done
+                if (namespaces.Count == 0)
+                {
+                    break;
+                }
+
+                // Skip using directives 
+                var usingDirectiveNamespace = usingDirective.Statement.ParsedNamespace;
+                if (usingDirectiveNamespace.StartsWith("System") || usingDirectiveNamespace.Contains("="))
+                {
+                    continue;
+                }
+
+                // Insert all using directives that fit before the next using
+                if (namespaces.Peek().CompareTo(usingDirectiveNamespace) < 0)
+                {
+                    var usingDirectiveLineIndex = codeDocument.Source.Lines.GetLocation(usingDirective.Node.Span.Start).LineIndex;
+                    var head = new Position(usingDirectiveLineIndex, 0);
+                    var edit = new TextEdit() { Range = new Range(head, head), NewText = "" };
+                    do
+                    {
+                        edit.NewText += $"@using {namespaces.Dequeue()}{Environment.NewLine}";
```

Comment:
instead of deuqueing inline I'd just use a local variable and then dequeue on its own line

Reply:
Yeah, my rule of thumb for this kind of thing is that if the state is going to be altered it should happen on its own line so that it's super explicit.